client/sites-dashboard/components/sites-table-row.tsx

```diff
@@ -142,7 +147,23 @@ export default function SitesTableRow( { site }: SiteTableRowProps ) {
 							href={ getDashboardUrl( site.slug ) }
 							title={ __( 'Visit Dashboard' ) }
 						>
-							<SiteIcon siteId={ site.ID } size={ 50 } />
+							<SiteThumbnail
+								mShotsUrl={ shouldUseScreenshot ? site.URL : undefined }
+								alt={ site.name }
+								bgColorImgUrl={ site.icon?.img }
+							>
+								{ site.icon ? (
+									<Image
+										src={ site.icon.img }
+										alt={ __( 'Site Icon' ) }
+										style={ { height: '50px', width: '50px' } }
+									/>
+								) : (
+									<NoIcon role={ 'img' } aria-label={ __( 'Site Icon' ) }>
+										{ site.name.charAt( 0 ) }
```

Comment:
I'm not sure this is 100% localisation friendly, but it'll probably be ok for now.

I think it'll work ok for words that start with simple diacritic mark. `site.name === Ā` for instance works fine.

This emoji for instance doesn't work (I guess I would have expected the icon to have been the complete emoji)
<img width="319" alt="CleanShot 2022-08-08 at 20 00 14@2x" src="https://user-images.githubusercontent.com/1500769/183369222-74d65216-3634-4e0b-b0b7-34b4c704e7c4.png">

Maybe emoji aren't the most important use case, but it might mean that there are non-latin languages where things don't work correctly. This example in Thai for instance:
<img width="254" alt="CleanShot 2022-08-08 at 19 58 48@2x" src="https://user-images.githubusercontent.com/1500769/183368990-e6d122e0-737a-466c-931f-358042880f90.png">
If you zoom in you can see that part of the first "character" (or grapheme) has been cut out of the icon. And I'm not sure what the significance of that will be.

Anyway, I guess we should just merge as is, but be aware this might need to change.

Reply:
Looks like the built-in `Intl` API might be able to help us
panfyZ-JL-p2_string-functions/#comment-1918

Reply:
The `Intl.Segmenter` type does it 👍 But it's not available on Firefox so we'd have to fallback to `charAt(0)` 😭

```typescript
function getFirstGrapheme( input: string ) {
	try {
		const segmenter = new Intl.Segmenter();
		const segments = segmenter.segment( input );
		return Array.from( segments )[0].segment;
	} catch {
		// Intl.Segmenter is not available in all browsers
		return input.charAt( 0 );
	}
}
```